src/foam/nanos/auth/CapabilityAuthService.js

```diff
@@ -0,0 +1,134 @@
+foam.CLASS({
+  package: 'foam.nanos.auth',
+  name: 'CapabilityAuthService',
+  extends: 'foam.nanos.auth.ProxyAuthService',
+  documentation: `
+  this decorator checks for either a capability or permission string. If the check returns false, delegate to next authservice. Return true otherwise.
+  `,
+
+  implements: [
+    'foam.nanos.auth.AuthService',
+  ],
+
+  javaImports: [
+    'foam.dao.ArraySink',
+    'foam.dao.DAO',
+    'foam.nanos.crunch.Capability',
+    'foam.nanos.crunch.CapabilityJunctionStatus',
+    'foam.nanos.crunch.UserCapabilityJunction',
+    'foam.nanos.logger.Logger',
+    'foam.nanos.session.Session',
+    'java.util.List',
+    'javax.security.auth.AuthPermission',
+    'static foam.mlang.MLang.*'
+  ],
+
+  imports: [
+
+  ],
+
+  methods: [
+    {
+      name: 'check',
+      documentation: `
+      check if the given input string is in the userCapabilityJunctions or implied by a capability in userCapabilityJunctions for the current context user
+      `,
+      javaCode: `
+      if ( x == null || permission == null ) return false;
```

Comment:
Is this the same code as checkUser()? Could we just call that method with the user taken out of the context?

Reply:
true, but I was wondering would there ever be a case where we want to call checkuser on a specific user instead of user in context?